client/blocks/app-banner/index.jsx

```diff
@@ -71,6 +73,15 @@ export class AppBanner extends Component {
 		}
 	}
 
+	loadDismissTimesExperiment() {
+		// Set a default value just in case the assignment hasn't loaded yet
+		this.experimentIsControl = true;
+
+		loadExperimentAssignment( APP_BANNER_EXPERIMENT_NAME ).then( ( experimentObject ) => {
```

Comment:
Since we're in a React component, we recommend using `loadExperimentAssignment` only for prefetching experiment data, which in this case may be done directly in `client/layout/index.jsx` where `AppBanner` component is also loaded asynchronously.

Then, the [React helper ProvideExperimentData](https://github.com/Automattic/wp-calypso/tree/trunk/packages/explat-client-react-helpers#api-provideexperimentdata) can be used to wrap what is returned by the `render` function of this class. Also, the `isLoading` argument of the render props function can be used to stop rendering the app banner before the experiment assignment data is loaded. Otherwise, there's a possible source of error with users assigned to `treatment` variant but [getting the control experience](https://github.com/Automattic/wp-calypso/pull/60830/files#diff-4e9725a52a528391bb297acf82b43ab022f6b0c4f47f6ec705140585196c1faeR77) because they've dismissed the banner too fast.

Reply:
Thanks for the review @razvanpapadopol I created a new PR with this change: https://github.com/Automattic/wp-calypso/pull/61047